server.js

```diff
@@ -52,12 +62,25 @@ app.get('/api/v1/foods', (request, response) => {
 })
 
 app.post('/api/v1/foods', (request, response) => {
-  Food.create(request.body.name, request.body.calories)
+  let diaryName = request.body.diary_name
+  let diaryData
+  // console.log();
+  Diary.findByName(diaryName)
+  .then((data) => {
+    diaryData = data
+
+  })
   .then(() => {
+    Food.create(request.body.name, request.body.calories)
+  .then(function(data) {
     Food.last()
-    .then(food => {
+  .then((food) => {
+    DiaryFood.create(diaryData.rows[0].id, food.rows[0].id)
+  .then( (data) => {
       response.json(food.rows[0])
-    })
+  })
```

Comment:
I know I typed this unindented form, I forgot to say it was more for debugging, with all those `})` at the same level of indentation, I'd move back to indenting them so it looks cleaner.